In [3]:
import pandas as pd

# Load your dataset from the correct path
file_path = r"C:\Users\elean\OneDrive - University of Virginia\Documents\DS 3001\Midterm\absolute-number-of-deaths-from-ambient-particulate-air-pollution.csv"

df = pd.read_csv(file_path)

# Preview the first few rows
df.head()


Entity Code  Year  \
0  Afghanistan  AFG  1990   
1  Afghanistan  AFG  1991   
2  Afghanistan  AFG  1992   
3  Afghanistan  AFG  1993   
4  Afghanistan  AFG  1994   

   Deaths - Cause: All causes - Risk: Outdoor air pollution - OWID - Sex: Both - Age: All Ages (Number)  \
0                                               3169                                                      
1                                               3222                                                      
2                                               3395                                                      
3                                               3623                                                      
4                                               3788                                                      

   Deaths - Cause: All causes - Risk: High systolic blood pressure - Sex: Both - Age: All Ages (Number)  \
0                                              25633                                                      
1                                              25872                                                      
2                                              26309                                                      
3                                              26961                                                      
4                                              27658                                                      

   Deaths - Cause: All causes - Risk: Diet high in sodium - Sex: Both - Age: All Ages (Number)  \
0                                               1045                                             
1                                               1055                                             
2                                               1075                                             
3                                               1103                                             
4                                               1134                                             

   Deaths - Cause: All causes - Risk: Diet low in whole grains - Sex: Both - Age: All Ages (Number)  \
0                                               7077                                                  
1                                               7149                                                  
2                                               7297                                                  
3                                               7499                                                  
4                                               7698                                                  

   Deaths - Cause: All causes - Risk: Alcohol use - Sex: Both - Age: All Ages (Number)  \
0                                                356                                     
1                                                364                                     
2                                                376                                     
3                                                389                                     
4                                                399                                     

   Deaths - Cause: All causes - Risk: Diet low in fruits - Sex: Both - Age: All Ages (Number)  \
0                                               3185                                            
1                                               3248                                            
2                                               3351                                            
3                                               3480                                            
4                                               3610                                            

   Deaths - Cause: All causes - Risk: Unsafe water source - Sex: Both - Age: All Ages (Number)  \
0                                               3702                                             
1                  

In [4]:
# See the structure of your dataset
df.info()

# Look at column names
print(df.columns.tolist())

# Check if there are missing values
df.isnull().sum()

# View a sample of rows
df.sample(5)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6840 entries, 0 to 6839
Data columns (total 31 columns):
 #   Column                                                                                                            Non-Null Count  Dtype 
---  ------                                                                                                            --------------  ----- 
 0   Entity                                                                                                            6840 non-null   object
 1   Code                                                                                                              6150 non-null   object
 2   Year                                                                                                              6840 non-null   int64 
 3   Deaths - Cause: All causes - Risk: Outdoor air pollution - OWID - Sex: Both - Age: All Ages (Number)              6840 non-null   int64 
 4   Deaths - Cause: All causes - Risk: H

Entity Code  Year  \
4656    Philippines  PHL  1996   
6761          Yemen  YEM  2001   
2779        Ireland  IRL  2009   
6533        Vietnam  VNM  2013   
1324  Cote d'Ivoire  CIV  1994   

      Deaths - Cause: All causes - Risk: Outdoor air pollution - OWID - Sex: Both - Age: All Ages (Number)  \
4656                                              14118                                                      
6761                                               4569                                                      
2779                                                799                                                      
6533                                              35595                                                      
1324                                               2938                                                      

      Deaths - Cause: All causes - Risk: High systolic blood pressure - Sex: Both - Age: All Ages (Number)  \
4656                                              39865                                                      
6761                                              17141                                                      
2779                                               5316                                                      
6533                                             145330                                                      
1324                                               8400                                                      

      Deaths - Cause: All causes - Risk: Diet high in sodium - Sex: Both - Age: All Ages (Number)  \
4656                                              10589                                             
6761                                                725                                             
2779                                                308                                             
6533                                              28720                                             
1324                                                826                                             

      Deaths - Cause: All causes - Risk: Diet low in whole grains - Sex: Both - Age: All Ages (Number)  \
4656                                               5131                                                  
6761                                               5013                                                  
2779                                                865                                                  
6533                                               9850                                                  
1324                                               1165                                                  

      Deaths - Cause: All causes - Risk: Alcohol use - Sex: Both - Age: All Ages (Number)  \
4656                                              23620                                     
6761                                                788                                     
2779                                               1530                                     
6533                                              43220                                     
1324                                               4607                                     

      Deaths - Cause: All causes - Risk: Diet low in fruits - Sex: Both - Age: All Ages (Number)  \
4656                                               3385                                            
6761                                               2106                                            
2779                                                446                                            
6533                                              14222                                            
1324                                                646                                            

      Deaths - Cause: All causes - Risk: Unsafe water source - Sex: Both - Age: Al

The dataset contains 6,840 rows and 31 columns. Only the ‘Code’ column has some missing values, likely for aggregated regions such as ‘World’ or continents. All other variables are complete.

In [11]:
# Step 3 (continued): Rename the already-shortened columns to use the "Deaths_due_to_" format

df = df.rename(columns={
    "Outdoor_Air_Pollution": "Deaths_due_to_Outdoor_Air_Pollution",
    "High_Blood_Pressure": "Deaths_due_to_High_Blood_Pressure",
    "High_Sodium_Diet": "Deaths_due_to_High_Sodium_Diet",
    "Low_Whole_Grains": "Deaths_due_to_Low_Whole_Grains",
    "Alcohol_Use": "Deaths_due_to_Alcohol_Use",
    "Low_Fruit_Diet": "Deaths_due_to_Low_Fruit_Diet",
    "Unsafe_Water": "Deaths_due_to_Unsafe_Water",
    "Secondhand_Smoke": "Deaths_due_to_Secondhand_Smoke",
    "Low_Birth_Weight": "Deaths_due_to_Low_Birth_Weight",
    "Child_Wasting": "Deaths_due_to_Child_Wasting",
    "Unsafe_Sex": "Deaths_due_to_Unsafe_Sex",
    "Low_Nuts_Seeds": "Deaths_due_to_Low_Nuts_Seeds",
    "Household_Air_Pollution": "Deaths_due_to_Household_Air_Pollution",
    "Low_Vegetables": "Deaths_due_to_Low_Vegetables",
    "Low_Physical_Activity": "Deaths_due_to_Low_Physical_Activity",
    "Smoking": "Deaths_due_to_Smoking",
    "High_Blood_Glucose": "Deaths_due_to_High_Blood_Glucose",
    "Air_Pollution_Total": "Deaths_due_to_Air_Pollution_Total",
    "High_BMI": "Deaths_due_to_High_BMI",
    "Unsafe_Sanitation": "Deaths_due_to_Unsafe_Sanitation",
    "No_Handwashing": "Deaths_due_to_No_Handwashing",
    "Drug_Use": "Deaths_due_to_Drug_Use",
    "Low_Bone_Density": "Deaths_due_to_Low_Bone_Density",
    "Vitamin_A_Deficiency": "Deaths_due_to_Vitamin_A_Deficiency",
    "Child_Stunting": "Deaths_due_to_Child_Stunting",
    "Discontinued_Breastfeeding": "Deaths_due_to_Discontinued_Breastfeeding",
    "NonExclusive_Breastfeeding": "Deaths_due_to_NonExclusive_Breastfeeding",
    "Iron_Deficiency": "Deaths_due_to_Iron_Deficiency"
})

# ✅ Double-check that it worked
print(df.columns.tolist())


['Country', 'Country_Code', 'Year', 'Outdoor_Air_Pollution_Deaths', 'Deaths_due_to_High_Blood_Pressure', 'Deaths_due_to_High_Sodium_Diet', 'Deaths_due_to_Low_Whole_Grains', 'Deaths_due_to_Alcohol_Use', 'Deaths_due_to_Low_Fruit_Diet', 'Deaths_due_to_Unsafe_Water', 'Deaths_due_to_Secondhand_Smoke', 'Deaths_due_to_Low_Birth_Weight', 'Deaths_due_to_Child_Wasting', 'Deaths_due_to_Unsafe_Sex', 'Deaths_due_to_Low_Nuts_Seeds', 'Deaths_due_to_Household_Air_Pollution', 'Deaths_due_to_Low_Vegetables', 'Deaths_due_to_Low_Physical_Activity', 'Deaths_due_to_Smoking', 'Deaths_due_to_High_Blood_Glucose', 'Deaths_due_to_Air_Pollution_Total', 'Deaths_due_to_High_BMI', 'Deaths_due_to_Unsafe_Sanitation', 'Deaths_due_to_No_Handwashing', 'Deaths_due_to_Drug_Use', 'Deaths_due_to_Low_Bone_Density', 'Deaths_due_to_Vitamin_A_Deficiency', 'Deaths_due_to_Child_Stunting', 'Deaths_due_to_Discontinued_Breastfeeding', 'Deaths_due_to_NonExclusive_Breastfeeding', 'Deaths_due_to_Iron_Deficiency']
